src/transformStateWithClones.js

```diff
@@ -7,7 +7,28 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const copyState = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    if (action.type === 'clear') {
+      for (const key of Object.keys(copyState)) {
+        delete copyState[key];
+      }
+    } else if (action.type === 'removeProperties') {
+      for (const key of action.keysToRemove) {
+        if (copyState[key]) {
+          delete copyState[key];
+        }
+      }
+    } else if (action.type === 'addProperties') {
+      Object.assign(copyState, action.extraData);
+    }
+
+    result.push({ ...copyState });
+  }
```

Comment:
Try to avoid nested if else statements, in this case better to use `switch/case`

Reply:
I tried to make a better solution, please check